### PR TITLE
Direct use of sed|awk commands

### LINKED ISSUE #458: ci: v0.14.0 Unable to build .deb package
If eza fails to compile, or if there is a problem during the build process, then please include the following information in your report:

- The exact eza commit you are building : **9a9df550**
- Your operating system and hardware platform : Docker debian/bookworm

If you are seeing compilation errors, please include the output of the build process.

`./devtools/deb-package.sh` output :
```
checkout tag v0.14.0
build man pages
sh: 1: Syntax error: "(" unexpected
error: Recipe `man` failed on line 80 with exit code 2
```
---

To reproduce :
``` sh
docker run -it "debian:bookworm" bash

apt-get update -yq && apt-get install -yq aptitude curl git wget gzip lintian pandoc gpg lsb-release

# Just
wget -qO - 'https://proget.makedeb.org/debian-feeds/prebuilt-mpr.pub' | gpg --dearmor | tee /usr/share/keyrings/prebuilt-mpr-archive-keyring.gpg 1> /dev/null
echo "deb [arch=all,$(dpkg --print-architecture) signed-by=/usr/share/keyrings/prebuilt-mpr-archive-keyring.gpg] https://proget.makedeb.org prebuilt-mpr $(lsb_release -cs)" | tee /etc/apt/sources.list.d/prebuilt-mpr.list
apt-get update -yq && aptitude --quiet --assume-yes install just

git clone https://github.com/eza-community/eza && cd eza

./devtools/deb-package.sh
```

### PR DESCRIPTION
**Pull request by gardouille**
Fix #458
+ remove extra whitespaces

### CODE REVIEW

**Comment by gardouille** [comment]
Woohoops… I haven't read the code of conduct, sorry :pray:

File: Justfile [hunk 1]
```diff
@@ -74,9 +74,9 @@ all-release: build-release test-release
 # build the man pages
 @man:
     mkdir -p "${CARGO_TARGET_DIR:-target}/man"
-    version=$(cat Cargo.toml | grep ^version | head -n 1 | awk '{print $NF}' | tr -d '"'); \
+    version=$(awk 'BEGIN { FS = "\"" } ; /^version/ { print $2 ; exit }' Cargo.toml); \
```
Comment (gierens) [inline Justfile:77]: Interesting, always good to learn some `awk` magic. ... Definitely makes this more elgant, but not sure it's more readable though, hmm.

File: Justfile [hunk 2]
```diff
@@ -74,9 +74,9 @@ all-release: build-release test-release
...
     for page in eza.1 eza_colors.5 eza_colors-explanation.5; do \
-        pandoc --standalone -f markdown -t man <(cat "man/${page}.md" | sed "s/\$version/v${version}/g") > "${CARGO_TARGET_DIR:-target}/man/${page}"; \
+        sed "s/\$version/v${version}/g" "man/${page}.md" | pandoc --standalone -f markdown -t man > "${CARGO_TARGET_DIR:-target}/man/${page}"; \
```
Comment (gierens) [inline Justfile:79]: That might be an improvement ... just wondering though, is there any other reason for this than older bash versions not having process substitution?

File: Justfile [hunk 3]
```diff
@@ -96,9 +96,9 @@ all-release: build-release test-release
 #---------------#
 
 # If you're not cafkafk and she isn't dead, don't run this!
-#
```
Comment (gierens) [inline Justfile:99]: The formatting fixes should ideally be part of a different commit.

**Review by gierens — CHANGES_REQUESTED** [review]
Thank's for the PR ... some interesting suggestions. I added a few questions and changes requests. Also the commit messages should follow conventional commit rules.

File: Justfile (hunk 3 above)
Comment (gardouille) [reply Justfile:99]: Yeah my bad, sorry for that.
My editor automatically remove extra whitespaces before saving :sweat_smile:

File: Justfile (hunk 2 above)
Comment (gardouille) [reply Justfile:79]: That's a good practice :)
https://github.com/koalaman/shellcheck/wiki/SC2002

File: Justfile (hunk 1 above)
Comment (gardouille) [reply Justfile:77]: Yeah, i wasn't sure that putting something too tricky was a good idea… But as that's the only call of `awk` command I replace everything by one "big" awk command.
I also think to something more readable : 
`awk '/^version/ { print $NF ; exit }' Cargo.toml | tr -d '"'`

But you can also reduce the elements with direct call to `grep` with the right option : 
`grep "^version" --max-count=1 Cargo.toml |  awk '{print $NF}' | tr -d '"'`

File: Justfile (hunk 3 above)
Comment (gierens) [reply Justfile:99]: git allows for splitting things into multiple commits retroactively ... like a soft reset, then commiting the changes individually and then force pushing here to overwrite the single commit.

Comment (gierens) [reply Justfile:99]: I mean we could also do that for you, but that will surely take longer.

File: Justfile (hunk 2 above)
Comment (gardouille) [reply Justfile:79]: And i forgot the most important…
Yes it works in bash, but not with `just` (at least for me…). Even a basic Justfile with the "man" fails :
```
cat Justfile
# build the man pages
@man:
    mkdir -p "${CARGO_TARGET_DIR:-target}/man"
    version=$(cat Cargo.toml | grep ^version | head -n 1 | awk '{print $NF}' | tr -d '"'); \
    for page in eza.1 eza_colors.5 eza_colors-explanation.5; do \
        pandoc --standalone -f markdown -t man <(cat "man/${page}.md" | sed "s/\$version/v${version}/g") > "${CARGO_TARGET_DIR:-target}/man/${page}"; \
    done;
```
```
just man
sh: 1: Syntax error: "(" unexpected
error: Recipe `man` failed on line 7 with exit code 2
```

just --version **1.14.0**

Comment (gierens) [reply Justfile:79]: I also use just v1.14.0 so that can't be the problem. ... As I said in the related issue my guess would be the bash version ... which one do you use?

Comment (gierens) [reply Justfile:79]: ah just saw your comment in the issue ... interesting

File: Justfile (hunk 1 above)
Comment (gierens) [reply Justfile:77]: No I think it's fine with just awk ... at least no one will ever attempt to change that ever again :D

File: Justfile (hunk 2 above)
Comment (gardouille) [reply Justfile:79]: Yeah, also sorry for the cross-posts…

Bash version : **5.2.15**.

And for the test, i put "build the man pages" commands to a tiny script shell : 
``` sh
cat pandoc.sh
#!/bin/bash -e

# build the man pages
rm -rf "${CARGO_TARGET_DIR:-target}/man"
mkdir -p "${CARGO_TARGET_DIR:-target}/man"
version=$(cat Cargo.toml | grep ^version | head -n 1 | awk '{print $NF}' | tr -d '"'); \
        for page in eza.1 eza_colors.5 eza_colors-explanation.5; do \
                pandoc --standalone -f markdown -t man <(cat "man/${page}.md" | sed "s/\$version/v${version}/g") > "${CARGO_TARGET_DIR:-target}/man/${page}"; \
done;
```

And… it works :sweat_smile: 
``` sh
./pandoc.sh

ls -l target/man/
total 28
-rw-r--r-- 1 root root 11486 Oct  2 14:45 eza.1
-rw-r--r-- 1 root root  2893 Oct  2 14:45 eza_colors-explanation.5
-rw-r--r-- 1 root root  8775 Oct  2 14:45 eza_colors.5
```

File: Justfile (hunk 3 above)
Comment (gardouille) [reply Justfile:99]: Hope it's better after spliting and rewriting commit messages :)

**Review by cafkafk — COMMENTED** [review]
The release part of the `justfile` will probably not be using this PR, see https://github.com/eza-community/eza/compare/main...cafk-better-release

**Review by gierens — APPROVED** [review]
I rebased this on the current main and checked that it works for me. Sorry for the late response. LGTM, ready for merge ;)

**Review by cafkafk — APPROVED** [review]
Cool, lgtm, ty for contributing

**Comment by gardouille** [comment]
Cool, thanks :+1:
